nodejs/awsx/apigateway/api.ts

```diff
@@ -605,37 +607,67 @@ function addAPIKeyToSwaggerOperation(swaggerOperation: SwaggerOperation) {
     swagger: SwaggerSpec,
     authorizers: Authorizer[],
-    apiAuthorizers: Record<string, Authorizer>): string[] {
+    apiAuthorizers: Record<string, Authorizer>): Record<string, string[]>[] {
 
-    const authNames: string[] = [];
+    const authRecords: Record<string, string[]>[] = [];
     swagger["securityDefinitions"] = swagger["securityDefinitions"] || {};
 
     for (const auth of authorizers) {
         const suffix = Object.keys(swagger["securityDefinitions"]).length;
-        auth.authorizerName = auth.authorizerName || `${swagger.info.title}-authorizer-${suffix}`;
+        const authName = auth.authorizerName || `${swagger.info.title}-authorizer-${suffix}`;
+        auth.authorizerName = authName;
 
         // Check API authorizers - if its a new authorizer add it to the apiAuthorizers
         // if the name already exists, we check that the authorizer references the same authorizer
-        if (!apiAuthorizers[auth.authorizerName]) {
-            apiAuthorizers[auth.authorizerName] = auth;
-        } else if (apiAuthorizers[auth.authorizerName] !== auth) {
-            throw new Error("Two different authorizers using the same name: " + auth.authorizerName);
+        if (!apiAuthorizers[authName]) {
+            apiAuthorizers[authName] = auth;
+        } else if (apiAuthorizers[authName] !== auth) {
+            throw new Error("Two different authorizers using the same name: " + authName);
         }
 
         // Add security definition if it's a new authorizer
         if (!swagger["securityDefinitions"][auth.authorizerName]) {
-            const securityDef: SecurityDefinition = {
+
+            swagger["securityDefinitions"][authName] = {
                 type: "apiKey",
                 name: auth.parameterName,
-                in: auth.parameterLocation,
-                "x-amazon-apigateway-authtype": auth.authType,
-                "x-amazon-apigateway-authorizer": getLambdaAuthorizer(auth.authorizerName, auth),
+                in: lambdaAuthorizer.isLambdaAuthorizer(auth) ? auth.parameterLocation : "header",
+                "x-amazon-apigateway-authtype": lambdaAuthorizer.isLambdaAuthorizer(auth) ? auth.authType : "cognito_user_pools",
+                "x-amazon-apigateway-authorizer": lambdaAuthorizer.isLambdaAuthorizer(auth)
+                    ? getLambdaAuthorizer(authName, auth)
+                    : getCognitoAuthorizer(auth.identitySource, auth.providerARNs),
             };
-            swagger["securityDefinitions"][auth.authorizerName] = securityDef;
         }
-        authNames.push(auth.authorizerName);
+
+        const methods = lambdaAuthorizer.isLambdaAuthorizer(auth) || !auth.methodsToAuthorize
+            ? []
+            : auth.methodsToAuthorize;
+
+        authRecords.push({ [authName]: methods });
+
     }
-    return authNames;
+    return authRecords;
+}
+
+function getCognitoPoolARNs(pools: Array<pulumi.Input<string> | aws.cognito.UserPool>): pulumi.Input<string>[] {
```

Comment:
nit: up to you  but return value is optional her.e  if you like it for documentation purposes, that's fine.

Reply:
If I dont assign it here then the compiler complains that:

Type 'string' is not assignable to type '"cognito_user_pools"'.